components/dot_menu/dot_menu.jsx

```diff
@@ -218,16 +218,19 @@ export default class DotMenu extends React.PureComponent {
         </Tooltip>
     )
 
-    refCallback = (ref) => {
-        if (ref) {
-            const rect = ref.rect();
-            const y = rect.y || rect.top;
+    refCallback = (menuRef) => {
+        if (menuRef) {
+            const rect = menuRef.rect();
+            const buttonRef = this.buttonRef.current.getBoundingClientRect();
+            const y = buttonRef.y || buttonRef.top;
             const height = rect.height;
             const windowHeight = window.innerHeight;
 
-            if ((y + height) > (windowHeight - MENU_BOTTOM_MARGIN)) {
-                this.setState({openUp: true});
-            }
+            const totalSpace = windowHeight - MENU_BOTTOM_MARGIN;
+            const spaceOnTop = y;
+            const spaceOnBottom = (totalSpace - (spaceOnTop - height));
```

Comment:
Can you check if `popOverOverlayPosition` in `utils/position_utils.tsx` serve similar purpose for this?

Reply:
Never mind, i will leave it upto other devs for review on this PR.